flake.nix

```diff
@@ -0,0 +1,32 @@
+{
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    naersk.url = "github:nix-community/naersk";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs = { self, flake-utils, naersk, nixpkgs }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = (import nixpkgs) {
+          inherit system;
+        };
+
+        naersk' = pkgs.callPackage naersk {};
+      in rec {
+        defaultPackage = naersk'.buildPackage {
```

Comment:
realistically, this will break in time unless it's somehow tested in CI. Any chance you could add a gitlab action which runs nix flake check or similar (and ensure that defaultPackage is built in this case)?

Could use https://github.com/DeterminateSystems/magic-nix-cache-action to ensure that the rebuild cost is low...